deps/wazuh_testing/wazuh_testing/scripts/statistical_data_analyzer.py

```diff
@@ -0,0 +1,339 @@
+                'Metric': value,
+                'Mean': round(float(daemon_data[value].mean()), 2),
+                'Median': round(float(daemon_data[value].median()), 2),
+                'Max value': round(float(daemon_data[value].max()), 2),
+                'Min value': round(float(daemon_data[value].min()), 2),
+                'Standard deviation': round(float(daemon_data[value].std()), 2),
+                'Variance': round(float(daemon_data[value].var()), 2)
+            }
+            stats.append(stat)
+        results[daemon] = stats
+    
+    return results
+
+
+def get_stat_value(metrics, value, stat):
+    """Checks if the metric is correct to return the specific statistic.
+
+    Args:
+        metrics: the metrics of a specific daemon.
+        value: value of this metric.
+        stat: statistics of this metric to be returned.
+    
+    Returns:
+        stat_value: value of the statistic desired.
+    """
+    for metric in metrics:
+        if metric['Metric'] == value:
+            return metric[stat]
+
+
+def comparison_basic_statistics(baseline, datasource, daemon, value, stat, threshold):
+    """Compares the percentage change in a given statistic between the two data sets, and 
+    returns whether there is a significant change based on a threshold value.
+
+    Args:
+        baseline: Dataframe with the baseline data.
+        datasource: Dataframe with the data source.
+        daemon: concrete daemon on which to obtain the values.
+        value: metric from which the statistics to be compared are obtained.
+        stat: concrete statistic to be compared.
+        threshold: Threshold for comparison. If not specified, default is 5%.
+
+    Returns:
+        discrepancie: If the percentage difference is greater than the threshold,
+        it returns 1, otherwise it returns 0.
+    """
+    baseline_statistics = calculate_basic_statistics(baseline)
+    dataframe_statistics = calculate_basic_statistics(datasource)
+    discrepancie = 0
+    baseline_value = get_stat_value(baseline_statistics[daemon], value, stat)
+    dataframe_value = get_stat_value(dataframe_statistics[daemon], value, stat)
+
+    if baseline_value != 0:
+        diff = abs(baseline_value - dataframe_value) / baseline_value
+    else:
+        diff = abs(baseline_value - dataframe_value)
+    
+    if diff >= threshold:
+        discrepancie = 1
+    
+    return discrepancie
+
+
+def t_student_test(baseline, datasource, value):
+    """Function that performs the statistical analysis using the t-student test.
+
+    Args:
+        baseline: Dataframe with the baseline data.
+        datasource: Dataframe with the data source.
+        value: metric on which the test is performed
+
+    Returns:
+        t_p_value: p value returned by the t-student test.
+    """
+    ref_values = baseline[value].dropna()
+    new_values = datasource[value].dropna() 
+
+    t_stat, t_p_value = ttest_ind(ref_values, new_values, equal_var=False)
```

Comment:
Why we don´t take into consideration the `t_stat`?

Reply:
This variable gives information to the statistical test (its value is used to determine how different, for example, the variance may be), but it is not used to detect significant changes between two sets of data; only the p_value is used for that purpose. Therefore, it is not taken into account in our tests.